app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -374,16 +385,27 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     }
 
     /**
-     * Share the playlist as a newline-separated list of stream URLs.
+     * Share the playlist as a newline-separated list of stream URLs and video names.
      */
-    public void sharePlaylist() {
+    private void sharePlaylist() {
+        final boolean shouldSharePlaylistDetails = sharedPreferences.getBoolean(
+                requireContext().getString(R.string.share_playlist_with_details_key), false);
         disposables.add(playlistManager.getPlaylistStreams(playlistId)
                 .flatMapSingle(playlist -> Single.just(playlist.stream()
                         .map(PlaylistStreamEntry::getStreamEntity)
-                        .map(StreamEntity::getUrl)
+                        .map(streamEntity -> {
+                            if (shouldSharePlaylistDetails) {
+                                return String.format("- %s: %s",
```

Comment:
I think we will need to allow the localization of this string for LTR countries. Please extract it into a string res.